pandas/tests/test_reshape.py

```diff
@@ -326,6 +326,20 @@ def test_include_na(self):
                                 dtype=np.uint8)
         tm.assert_numpy_array_equal(res_just_na.values, exp_just_na.values)
 
+    def test_keep_na(self):
+        s = ['a', 'b', np.nan]
+        res_na = get_dummies(s, dummy_na='keep', sparse=self.sparse)
+        exp_na = DataFrame({'a': {0: 1, 1: 0, 2: np.nan},
+                            'b': {0: 0, 1: 1, 2: np.nan}},
+                           dtype=np.float16)
```

Comment:
@jreback have you added a `pytest.mark` for marking stuff that will change in pandas2? these "should" have int dtype in pandas 2.